odd_collector/adapters/databricks/mappers/column.py

```diff
@@ -11,18 +17,134 @@ def split_by_braces(value: str) -> str:
+                    oddrn=oddrn,
+                    name=column_name,
+                    metadata=[
+                        extract_metadata(
+                            "databricks", column, DefinitionType.DATASET_FIELD
+                        )
+                    ],
+                    type=DataSetFieldType(
+                        type=Type.TYPE_STRUCT,
+                        logical_type=get_logical_type(column_type),
+                        is_nullable=False,
+                    ),
+                    owner=None,
+                    parent_field_oddrn=parent_oddrn,
+                )
+            )
+            for field_name, field_type in column_type.fields.items():
+                _build_ds_field_from_type(field_name, field_type, oddrn)
+        elif isinstance(column_type, Map):
+            generated_dataset_fields.append(
+                DataSetField(
+                    oddrn=oddrn,
+                    name=column_name,
+                    metadata=[
+                        extract_metadata(
+                            "databricks", column, DefinitionType.DATASET_FIELD
+                        )
+                    ],
+                    type=DataSetFieldType(
+                        type=Type.TYPE_MAP,
+                        logical_type=get_logical_type(column_type),
+                        is_nullable=False,
+                    ),
+                    owner=None,
+                    parent_field_oddrn=parent_oddrn,
+                )
+            )
+            _build_ds_field_from_type("Key", column_type.key_type, oddrn)
+            _build_ds_field_from_type("Value", column_type.value_type, oddrn)
+
+        else:
+            odd_type = get_databricks_type(column_type)
+            logical_type = get_logical_type(column_type)
+            logger.debug(
+                f"Column {column_name} has ODD type {odd_type} and logical type {logical_type}"
+            )
+            generated_dataset_fields.append(
+                DataSetField(
+                    oddrn=oddrn,
+                    name=column_name,
+                    metadata=[
+                        extract_metadata(
+                            "databricks", column, DefinitionType.DATASET_FIELD
+                        )
+                    ],
+                    type=DataSetFieldType(
+                        type=odd_type,
+                        logical_type=logical_type,
+                        is_nullable=False,
+                    ),
+                    owner=None,
+                    parent_field_oddrn=parent_oddrn,
+                )
+            )
+
+    _build_ds_field_from_type(column.name, column_type)
+    return generated_dataset_fields
+
+
+def get_logical_type(type_field: Union[ParseType, str]) -> str:
+    if isinstance(type_field, BasicType):
+        return type_field.type_name
+    elif isinstance(type_field, ArrayType):
+        return f"Array({get_logical_type(type_field.type)})"
```

Comment:
Looks like you have created method `to_databricks_type` for that purposes

Reply:
This function helps to construct logical types.
`to_databricks_type()` returns Type instance. It will be renamed to `to_odd_type()`